f3/api.go

```diff
@@ -47,15 +47,23 @@ type Clock interface {
 type Signer interface {
 	// Signs a message for the given sender ID.
 	Sign(sender ActorID, msg []byte) []byte
+}
+
+type Verifier interface {
 	// Verifies a signature for the given sender ID.
-	Verify(sender ActorID, msg, sig []byte) bool
+	Verify(pubKey PubKey, msg, sig []byte) bool
+	// Aggregates signatures from a participant to an existing signature.
+	Aggregate(sig [][]byte, aggSignature []byte) []byte
```

Comment:
I see that you are assuming that aggregate signatures can be combined. I think the conclusion is that it is not true, but we can fix that later.